index.js

```diff
@@ -59,13 +61,23 @@ function lint(input, config, webpack) {
     res = engine.executeOnText(input, resourcePath, true)
 
     // Save new results in the cache
-    if (config.cache) {
+    if (config.cache && cachePath) {
       cache[resourcePath] = {
         hash: inputMD5,
         rules: rulesHash,
         res: res,
       }
-      fs.writeFileSync(cachePath, JSON.stringify(cache))
+      var cacheJson = JSON.stringify(cacheJson)
+      try {
+        safeWriteCache(cachePath)
```

Comment:
The function definition says it accepts `cacheJson` but you're passing `cachePath`.
This is a bit confusing—it's hard to tell what is local and what is global, and why.
Can you avoid shared mutable state wherever possible?